core/templates/pages/contributor-dashboard-page/services/contribution-opportunities-backend-api.service.ts

```diff
@@ -215,6 +215,29 @@ export class ContributionOpportunitiesBackendApiService {
       return [];
     }
   }
+
+  async saveUserLanguagePreferenceAsync(language_code: string):
+  Promise<unknown> {
+    const postData: {'language_code': string} = {
+      language_code: language_code,
+    }
+    const body = new FormData();
+    body.append('payload', JSON.stringify(postData));
+    const response = this.http.post(
+      '/prefferedtranslationlanguage', body).toPromise();
+    return response;
+  }
+
+  async fetchUserLanguagePreferenceAsync():
+  Promise<string|null> {
+    try {
+      const response = await this.http.get<string>(
+        '/prefferedtranslationlanguage').toPromise();
+      return response;
+    } catch {
+      return null;
```

Comment:
What exception is being caught here?

Reply:
I wrote this in a try-catch block to deal with errors during this promise. I have now changed the way a little bit. Kindly, have another look.

Reply:
I have changed this back to what it was because our code might break if we throw an error over here. Actually, I am following the general pattern over here which is being followed elsewhere in this file.
E.g. lines 191-218 https://github.com/oppia/oppia/blob/02db474efed7bfb91864c04ba3db2c82969fb5aa/core/templates/pages/contributor-dashboard-page/services/contribution-opportunities-backend-api.service.ts#L191-L218